### PR TITLE
Avoid infinite recursion in GetBitSize/GetBitAlign

### PR DESCRIPTION
**Pull request by adrian-prantl**
rdar://79874370

### CODE REVIEW

**Comment by adrian-prantl** [comment]
@swift-ci test

File: lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp [hunk 1]
```diff
@@ -2170,18 +2170,32 @@ TypeSystemSwiftTypeRef::GetBitSize(opaque_compiler_type_t type,
...
 
     // If there is no process, we can still try to get the static size
     // information out of DWARF. Because it is stored in the Type
     // object we need to look that up by name again.
     if (TypeSP type_sp = LookupTypeInModule(type)) {
-      if (auto byte_size = type_sp->GetByteSize(exe_scope))
+      struct SwiftType : public Type {
+        /// Avoid a potential infinite recursion because
+        /// Type::GetByteSize() may call into this function again.
+        llvm::Optional<uint64_t> GetStaticByteSize() {
```
Comment (kastiglione) [inline lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp:2186]: Thoughts on adding this function directly to `Type`?

File: lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp [hunk 2]
```diff
@@ -3179,9 +3193,10 @@ TypeSystemSwiftTypeRef::GetTypeBitAlign(opaque_compiler_type_t type,
   // alignment information out of DWARF. Because it is stored in the
   // Type object we need to look that up by name again.
   if (TypeSP type_sp = LookupTypeInModule(type))
-    return type_sp->GetLayoutCompilerType().GetTypeBitAlign(exe_scope);
+    if (type_sp->GetLayoutCompilerType().GetOpaqueQualType() != type)
```
Comment (kastiglione) [inline lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp:3196]: Suggestion: a comment saying when/why this inequality happens.

**Review by kastiglione — APPROVED** [review]

File: lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp (hunk 1 above)
Comment (adrian-prantl) [reply lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp:2186]: I did it in this odd way is to minimize the diff between swift-lldb and llvm.org lldb.

File: lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp (hunk 2 above)
Comment (adrian-prantl) [reply lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp:3196]: It's again to avoid an infinite recursion. I'll add a comment.

**Pull request merged**